qa/tasks/cephfs/test_failover.py

```diff
@@ -706,6 +709,90 @@ def test_rank_stopped(self):
         status = self._confirm_single_replay()
         self.assertTrue(standby_count, len(list(status.get_standbys())))
 
+    def test_health_warn_oversize_cache_has_no_counters(self):
+        '''
+        Test that when MDS cache size crosses the limit, health warning
+        printed for standy-replay MDS doesn't include inode and stray
+        counters.
+
+        Tests: https://tracker.ceph.com/issues/63514
+        '''
+        # reduce MDS cache limit, default MDS cache limit is too high which
+        # will unnecessarily consume too many resources and too much time.
+        self.run_ceph_cmd('config set mds mds_cache_memory_limit 1K')
+        # health warning for crossing MDS cache size limit won't be raised
+        # until a threshold. default threshold is too high. it will
+        # unnecssarily consume so much time and resources.
+        self.run_ceph_cmd('config set mds mds_health_cache_threshold 1.000001')
+        # so that there is only active MDS and only 1 health warning is
+        # produced. presence of 2 warning should cause this test to fail
+        self.fs.set_max_mds(1)
+        self.run_ceph_cmd(f'fs set {self.fs.name} allow_standby_replay true')
+        self._confirm_single_replay()
+        self.fs.wait_for_daemons()
+        # The call above (to self.fs.wait_for_daemons()) should ensure we have
+        # only 1 active MDS on cluster
+        active_mds_id = self.fs.get_active_names()[0]
+        sr_mds_id = self.fs.get_standby_replay_names()[0]
+
+        # this should generate MDS cache more than 3M and therefore should
+        # trigger health warning MDS_CACHE_OVERSIZED.
+        open_proc = self.mount_a.open_n_background(".", 4000)
+
+        # actual test begins now...
+        errmsg = ('this test didn\'t generate health warning it was supposed '
+                  'to. health report has been logged above.')
+        with safe_while(sleep=3, tries=10, action=errmsg) as proceed:
+            while proceed():
+                # logging cache generated so far for th sake of easy
+                # debugging in future.
+                self.get_ceph_cmd_stdout(
+                    f'daemon mds.{active_mds_id} cache status')
+
+                health_report = self.get_ceph_cmd_stdout(
+                    'health detail --format json')
+                health_report = json_loads(health_report)
+                try:
+                    health_report['checks']['MDS_CACHE_OVERSIZED']
+                except KeyError:
+                    log.info('warning hasn\'t appeared in health report yet.'
```

Comment:
same as - https://github.com/ceph/ceph/pull/56065#discussion_r1562442844

Reply:
This is much needed info so that reason for repeatedly running `ceph health detail` is apparent in logs.

Reply:
Anyways, I've changed it so that this PR can proceed towards merge...

Reply:
> This is much needed info so that reason for repeatedly running `ceph health detail` is apparent in logs.

Right and it needs to be debug log level since it's used to debugging.

> Anyways, I've changed it so that this PR can proceed towards merge...

You should be really changing it if you are convinced and not to get it merged somehow (well, that's the goal, yes, but my aim **is not** to force people write patches that's only acceptable to me :)

Reply:
Okay. I thought you wanted this change to be incorporated. Therefore, I was willing to let it go. :)

Reply:
Yes, I did, but not due to the reason of getting it merged. I wanted to put forward the thought that just because the log level is required for vstart_runner, it isn't a justified reason to keep it at that log level.